datasquare_test.go

```diff
@@ -67,6 +67,28 @@ func TestRoots(t *testing.T) {
 	}
 }
 
+func TestLazyRootGeneration(t *testing.T) {
+	square, err := newDataSquare([][]byte{{1}, {2}, {3}, {4}}, NewDefaultTree)
+	if err != nil {
+		panic(err)
+	}
+
+	var rowRoots [][]byte
+	var colRoots [][]byte
+
+	for i := uint(0); i < 2; i++ {
+		rowRoots = append(rowRoots, square.RowRoot(i))
+		colRoots = append(rowRoots, square.ColRoot(i))
+	}
+
+	square.computeRoots()
```

Comment:
It would be good to have a test that only uses public API, too.

e.g.: assert that calling `RowRoots()[i]` yields the same result as calling `RowRoot(i)` etc

Reply:
good call. resolved https://github.com/lazyledger/rsmt2d/pull/18/commits/06294d583ac8f33a01085da3f578ba40f4151c2a